eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/data/TagManagerUtilities.java

```diff
@@ -0,0 +1,331 @@
+   * @return the corresponding tag provider tag name
+   * @since 1.0.0
+   */
+  public static String getTagNameForProviderFromDMWebEwon(
+      EwonConnectorSettings connectorSettings, DMWebEwon dmWebEwon, DMWebEwonTag dmWebEwonTag) {
+    // Get M2Web Ewon tag object
+    M2WebEwonEBDTag cachedM2WebEwonTag =
+        CacheManager.getCachedM2WebEwonTagMap(dmWebEwon.getName()).get(dmWebEwonTag.getName());
+
+    // Get tag name for provider
+    boolean isSystemTag = false;
+    String tagNameForProvider;
+    if (cachedM2WebEwonTag != null) {
+      tagNameForProvider =
+          getTagNameForProviderFromRawTagInfo(
+              connectorSettings,
+              dmWebEwon.getName(),
+              dmWebEwonTag.getName(),
+              getGroupBooleanFromInt(cachedM2WebEwonTag.getIvGroupA()),
+              getGroupBooleanFromInt(cachedM2WebEwonTag.getIvGroupB()),
+              getGroupBooleanFromInt(cachedM2WebEwonTag.getIvGroupC()),
+              getGroupBooleanFromInt(cachedM2WebEwonTag.getIvGroupD()),
+              isSystemTag);
+    } else {
+      // Show warning if debug enabled or if tag group sorting is enabled
+      if (connectorSettings.isSortTagsByGroup() || connectorSettings.isDebugEnabled()) {
+        LOGGER.warn(
+            "Could not find cached M2Web Ewon tag for the corresponding DMWeb Ewon tag: {}. Tag"
+                + " group sorting cannot be applied (if enabled), falling back to standard tag"
+                + " name!",
+            dmWebEwon.getName() + "/" + dmWebEwonTag.getName());
+      }
+      boolean isInGroup = false;
+      tagNameForProvider =
+          getTagNameForProviderFromRawTagInfo(
+              connectorSettings,
+              dmWebEwon.getName(),
+              dmWebEwonTag.getName(),
+              isInGroup,
+              isInGroup,
+              isInGroup,
+              isInGroup,
+              isSystemTag);
+    }
+    return tagNameForProvider;
+  }
+
+  /**
+   * Builds a {@link HistoricalTagValue} object from the specified tag provider and tag data point
+   * parameters.
+   *
+   * @param tagProviderName the name of the tag provider
+   * @param tagName the name of the tag
+   * @param tagDataType the {@link DataType} of the tag
+   * @param tagDataQuality the {@link DataQuality} of the tag value
+   * @param tagValueTimestamp the timestamp of the tag value
+   * @param tagValue the value of the tag
+   * @return the {@link HistoricalTagValue} object
+   * @since 1.0.0
+   */
+  public static HistoricalTagValue buildHistoricalTagValue(
+      String tagProviderName,
+      String tagName,
+      DataType tagDataType,
+      DataQuality tagDataQuality,
+      Date tagValueTimestamp,
+      Object tagValue) {
+    TagPath tagPath = TagPathParser.parseSafe(tagProviderName, tagName);
+    return new PackedHistoricalTagValue(
+        tagPath,
+        tagDataType.getTypeClass(),
+        tagDataType.getTypeClass() == DataTypeClass.Float
```

Comment:
I am not sure what the true impact of changing this would be. 

This functionality has existed since the original 7.9 and subsequent 8.x releases of the connector. It was added by an Inductive Automation employee when the connector was developed, and was under their control.

It appears to be related to how values are interpreted for deadband, as well as compression: https://docs.inductiveautomation.com/display/DOC79/Configuring+Tag+Historian#ConfiguringTagHistorian-ValueMode

Given the length that this has been in place, do you think it is best to leave it as is? That will keep the historical data functioning the same as in previous versions.

Reply:
As long as we have tested all Flexy tag types, leave as is.